shared/middleware/helpers/storage_apps.rb

```diff
@@ -74,7 +74,18 @@ def publish(channel_id, type)
     update_count = @table.where(id: id).exclude(state: 'deleted').update(row)
     raise NotFound, "channel `#{channel_id}` not found" if update_count == 0
 
-    row[:published_at]
+    project = @table.where(id: id).first
+    project_value = project[:value] ? JSON.parse(project[:value]) : {}
+    {
+      channel: channel_id,
+      name: project_value['name'],
+      thumbnailUrl: project_value['thumbnailUrl'],
+      type: project[:project_type],
+      publishedAt: project[:published_at],
+      # For privacy reasons, include only the first initial of the student's name.
+      studentName: user && UserHelpers.initial(user[:name]),
+      studentAgeRange: user && UserHelpers.age_range_from_birthday(user[:birthday]),
+    }
```

Comment:
This is a fair question. I ended up making this logic as shared as I could, working around the following obstacles:
1. the user data is mixed in with the project data via sequel join in one case, but separate in the other
2. some project fields are renamed during the sequel join